spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/TcpNioConnection.java

```diff
@@ -620,11 +625,27 @@ public void write(int b) throws IOException {
 
 		@Override
 		public void close() {
-			doClose();
+			if (this.selector != null) {
+				try {
+					this.selector.close();
+				}
+				catch (@SuppressWarnings(UNUSED) Exception e) {
+				}
+			}
+			try {
+				super.close();
+			}
+			catch (@SuppressWarnings(UNUSED) Exception e) {
+			}
 		}
 
 		@Override
 		public void flush() {
+			try {
+				super.flush();
```

Comment:
No reason to call `super`: it is just an empty method.

I think we can just remove this method altogether.

Reply:
no problem will not override flush at all